### PR TITLE
Add value sanity check and Len to DeltaTracker.

### PR DESCRIPTION
**Pull request by fasaxc**
## Description

Add a Len method to the Desired() side of the DeltaTracker.

Panic if trying to use a map in the DeltaTracker.  Found this was particularly hard to get right in another PR.

## Related issues/PRs

Enabling features for #8418

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
TBD
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/deltatracker/delta_tracker.go
Comment (nelljerram) [inline felix/deltatracker/delta_tracker.go:47]: Can you be more precise?  E.g. "It's not recommended to pass a mutable object such as <X> to the <Y> API(s)."

File: felix/deltatracker/delta_tracker.go
Comment (nelljerram) [inline felix/deltatracker/delta_tracker.go:50]: I think you're saying that the DeltaTracker doesn't perform copies (either shallow or deep).  Is the more important point that it doesn't copy the data when initially passed to it, or that it doesn't copy data when moving between Desired and Dataplane?  (Or both equally.)

File: felix/deltatracker/delta_tracker.go
Comment (nelljerram) [inline felix/deltatracker/delta_tracker.go:53]: I don't follow this.  Does DeltaTracker provide an API for "delete the object from one side of the tracker"?  If it does, surely that ought to be coded safely?

**Review by nelljerram — COMMENTED** [review]
I'm sorry to have taken a while before these initial questions, but I think I'm not properly understanding yet...

**Review by nelljerram — APPROVED** [review]

**Comment by marvin-tigera** [comment]
OK, I will merge the pull request when it's ready, leave the commits as is when I merge it, and leave the branch after I've merged it.

**Pull request merged**
